### PR TITLE
[feat] Add the ability to sync sliders of stacked boxes

### PR DESCRIPTION
**Pull request by KaroMourad**
- Add `sync` property to the `useDepthMap` custom hook
- Fix resetting depth map values on box config change

### CODE REVIEW

**Comment by KaroMourad** [comment]
Resolves #1731

**Review by roubkar — APPROVED** [review]

File: aim/web/ui/src/hooks/useDepthMap.tsx [hunk 1]
```diff
@@ -6,43 +6,59 @@ import _ from 'lodash-es';
...
 
 interface IUseDepthMapProps<T> {
   data: Array<T>;
   groupItemCb: (item: T) => string;
   state: any;
   deps: unknown[];
+  sync?: boolean;
 }
 
+type IUseDepthMap = [
```
Comment (rubenaprikyan) [inline aim/web/ui/src/hooks/useDepthMap.tsx:23]: Please use `I` for only interfaces

File: aim/web/ui/src/hooks/useDepthMap.tsx [hunk 2]
```diff
@@ -6,43 +6,59 @@ import _ from 'lodash-es';
...
+  const grouped = _.groupBy(data, groupItemCb);
+
+  const generateMapBy = React.useCallback((value: number = 0) => {
+    const newDepthMap: Record<string, number> = {};
+    for (let [groupId, items] of Object.entries(grouped)) {
+      const maxValue = items.length - 1;
+      newDepthMap[groupId] = value > maxValue ? maxValue : value;
+    }
+    return newDepthMap;
+  }, deps);
```
Comment (rubenaprikyan) [inline aim/web/ui/src/hooks/useDepthMap.tsx:44]: Why dependencies comes from outside?, you know everything `data`, `groupedItemCallback` in your memoized function

**Review by rubenaprikyan — CHANGES_REQUESTED** [review]

File: aim/web/ui/src/hooks/useDepthMap.tsx (hunk 2 above)
Comment (KaroMourad) [reply aim/web/ui/src/hooks/useDepthMap.tsx:44]: we don't need to update `generateMapBy` function every time when data changed as it only works when `deps` updates

**Review by rubenaprikyan — APPROVED** [review]
Lgtm!

**Pull request merged**